### DISCUSSION TITLE
Add Support for all LoRa configurations

### DISCUSSION THREAD

**Discussion by warrenb78** [Lora-net/LoRaMac-node#1566] [Ideas]
I am trying to create a DIY project using this library, and I don't want it to communicate over LoraWan, I just want simple LoRa. This library, that is also used in the Zephyr project, limits itself to LoRaWANs configurations, for example the supported LoRa BW is only one of the following:
* 125 KHz
* 250 KHz
* 500 KHz

But the LoRa phy supports much more configurations, starting at 7 KHz.
Why aren't they supported in this library?
Can you add support for all the phy's options in a future release?

**Comment by danak6jq** [comment]
I'm pretty sure this repo is intended for building LoRaWAN applications. If someone forks the library for non-LoRaWAN use, they might add support for smaller BW configurations. I doubt this support would be merged into this repo, but could be maintained separately.

Beware that smaller BW/longer packet time requires better frequency accuracy and stability; modules will require a TCXO.

Also beware that longer packet time may be limited by administrative restrictions (ex. 400mS in US915).

**Reply by mluis1** [reply]
The initial goal of this project was indeed to build LoRaWAN applications. The radio drivers were implemented with that goal in mind.

However, in [v5.0.0-branch](https://github.com/Lora-net/LoRaMac-node/tree/v5.0.0-branch) branch we have implemented new radio drivers which are able to handle all specific radio supported data rates.
